graylog2-web-interface/src/views/components/aggregationwizard/units/FieldUnitPopover.tsx

```diff
@@ -0,0 +1,148 @@
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useCallback, useMemo, useState } from 'react';
+import { styled, css } from 'styled-components';
+import { Field, useFormikContext } from 'formik';
+
+import Select from 'components/common/Select';
+import Popover from 'components/common/Popover';
+import { HoverForHelp, ModalButtonToolbar } from 'components/common';
+import { Alert, Button, Input } from 'components/bootstrap';
+import type { Unit } from 'views/components/visualizations/utils/unitConverters';
+import { mappedUnitsFromJSON as units } from 'views/components/visualizations/utils/unitConverters';
+import type { FieldUnitsFormValues } from 'views/types';
+import type FieldUnit from 'views/logic/aggregationbuilder/FieldUnit';
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: stretch;
+  & .control-label {
+    font-weight: normal;
+  }
+
+  width: 300px;
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  width: 100%;
+  justify-content: center;
+  align-items: center;
+  height: 25px;
+`;
+
+export const StyledButton = styled(Button)(({ theme }) => css`
+  background-color: ${theme.colors.gray[60]};
+  padding: 1px 2px;
+  min-width: 20px;
+  border-radius: 3px;
+  color: ${theme.colors.variant.lightest.default};
+  &:hover {
+    background-color: ${theme.colors.gray[80]};
+    color: ${theme.colors.variant.lightest.default};
+  }
+`);
+
+const FieldUnitPopover = ({ field, predefinedUnit }: { field: string, predefinedUnit: FieldUnit }) => {
+  const [show, setShow] = useState(false);
+  const { setFieldValue, values } = useFormikContext<{units: FieldUnitsFormValues }>();
+  const currentUnitType = useMemo<string>(() => values?.units?.[field]?.unitType, [values, field]);
+  const unitTypesOptions = useMemo(() => Object.keys(units).map((key) => ({ value: key, label: key })), []);
+  const unitOptions = useMemo(() => currentUnitType && units[currentUnitType]
+    .map(({ abbrev, name }: Unit) => ({ value: abbrev, label: name })), [currentUnitType]);
+  const toggleShow = () => setShow((cur) => !cur);
+  const onUnitTypeChange = useCallback((val: string) => {
+    setFieldValue(`units.${field}`, { unitType: val || undefined, abbrev: undefined });
+  }, [field, setFieldValue]);
+
+  const badgeLabel = useMemo(() => {
+    const curUnit = values?.units?.[field]?.abbrev;
+
+    return curUnit || '...';
+  }, [field, values?.units]);
+
+  const predefinedInfo = useMemo(() => {
+    if (!predefinedUnit?.isDefined) return null;
+
+    const unitName = units[predefinedUnit.unitType].find(({ abbrev }) => abbrev === predefinedUnit?.abbrev).name;
+
+    return <>Unit <b>{unitName}</b> was defined for field <b>{field}</b>by Graylog. Changing this unit might represent data incorrectly on the charts</>;
```

Comment:
We should insert a space here for better readability:
```suggestion
    return <>Unit <b>{unitName}</b> was defined for field <b>{field}</b> by Graylog. Changing this unit might represent data incorrectly on the charts</>;
```